keras_nlp/models/roberta.py

```diff
@@ -0,0 +1,298 @@
+            name=name,
+            trainable=trainable,
+        )
+        # All references to `self` below this line
+        self.token_embedding = embedding_layer.token_embedding
+        self.vocabulary_size = vocabulary_size
+        self.hidden_dim = hidden_dim
+        self.intermediate_dim = intermediate_dim
+        self.num_layers = num_layers
+        self.num_heads = num_heads
+        self.max_sequence_length = max_sequence_length
+        self.intermediate_dim = intermediate_dim
+        self.dropout = dropout
+        # BOS token '<s>' is equivalent to '[CLS]' from BERT
+        self.start_token_index = start_token_index
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "vocabulary_size": self.vocabulary_size,
+                "hidden_dim": self.hidden_dim,
+                "intermediate_dim": self.intermediate_dim,
+                "num_layers": self.num_layers,
+                "num_heads": self.num_heads,
+                "max_sequence_length": self.max_sequence_length,
+                "dropout": self.dropout,
+                "start_token_index": self.start_token_index,
+            }
+        )
+        return config
+
+
+# TODO: add RobertaMLM, different from BERT.
+
+
+class RobertaClassifier(keras.Model):
+    """Roberta encoder model with a classification head.
+
+    Args:
+        base_model: A `keras_nlp.models.Roberta` to encode inputs.
+        num_classes: Int. Number of classes to predict.
+        hidden_dim: Int. The size of the pooler layer.
+        name: String, optional. Name of the model.
+        trainable: Boolean, optional. If the model's variables should be
+            trainable.
+
+    Example usage:
+    ```
+    python
+    # Randomly initialized Roberta encoder
+    encoder = keras_nlp.models.Roberta(
+        vocabulary_size=50265,
+        num_layers=12,
+        num_heads=12,
+        hidden_dim=768,
+        intermediate_dim=3072,
+        max_sequence_length=12
+    )
+
+    # Call classifier on the inputs.
+    input_data = {
+        "input_ids": tf.random.uniform(
+            shape=(1, 12), dtype=tf.int64, maxval=50265),
+        "input_mask": tf.constant(
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0], shape=(1, 12)),
+    }
+    classifier = keras_nlp.models.RobertaClassifier(encoder, 4)
+    logits = classifier(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        base_model,
+        num_classes,
+        hidden_dim=None,
+        dropout=0.1,
+        name=None,
+        trainable=True,
+    ):
+        inputs = base_model.input
+        if hidden_dim is None:
+            hidden_dim = base_model.hidden_dim
+
+        x = base_model(inputs)["sequence_output"][
+            :, base_model.start_token_index, :
+        ]
+        x = keras.layers.Dropout(dropout, name="pooled_dropout")(x)
+        x = keras.layers.Dense(
+            hidden_dim, activation="tanh", name="pooled_dense"
```

Comment:
I assume you dug these implementation details out of some other implementation? Which one? The paper barely talking about fine tuning strategy let alone using `tanh` activation and all this dropout.

However, if this stands up to checkpoint conversion we are golden.

Reply:
So the Fairseq classifier head implementation is here, this demonstrates the dropouts:
https://github.com/facebookresearch/fairseq/blob/eda703798dcfde11c1ee517805c27e8698285d71/fairseq/models/roberta/model.py#L498
The Tanh activation is defined here, which gets propagated all the way to the Classifier head
https://github.com/facebookresearch/fairseq/blob/eda703798dcfde11c1ee517805c27e8698285d71/fairseq/models/roberta/model.py#L652

I'll do some checkpoint conversion testing for this to double check